engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2ServiceFactory.scala

```diff
@@ -1,19 +1,55 @@
 package pl.touk.nussknacker.ui.security.oauth2
 
-import org.asynchttpclient.DefaultAsyncHttpClientConfig
 import pl.touk.nussknacker.ui.security.api.LoggedUser
-import sttp.client.asynchttpclient.future.AsyncHttpClientFutureBackend
 import sttp.client.{NothingT, SttpBackend}
 
+import scala.concurrent.duration.{Deadline, FiniteDuration}
 import scala.concurrent.{ExecutionContext, Future}
 
+trait OAuth2AuthorizationData {
+  val accessToken: String
+  val expirationPeriod: Option[FiniteDuration]
+  val tokenType: String
+  val refreshToken: Option[String]
+}
+
+trait OAuth2NewService[+UserInfoData, +AuthorizationData <: OAuth2AuthorizationData] {
```

Comment:
I wonder if it makes sense to maintain backwards compatibility here. We'll be left with this name forever. For non-essential API we usually just add migration guide?

Reply:
True, but changing this will make the PR even bigger. :) I thought we could do it in two steps.

Reply:
I think I can change `OAuthService` into `OAuthOldService` and `OAuthNewService` into `OAuthService` (and do all the subsequent changes) so that the simplest migration would be to import the old class with an alias.

Reply:
:+1:

Reply:
Done. I've also put an entry in the migration guide.